src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -761,16 +755,17 @@ private async Task<long> ReplayTaggedMessagesAsync(ReplayTaggedMessages replay,
                         counter++;
                     }
 
-                    maxOrderingId = Math.Max(maxOrderingId, entry.UtcTicks);
+                    currentOrderingId = Math.Max(currentOrderingId, entry.UtcTicks);
+                    
+                    if (counter >= replay.Max)
+                        return new ReplayTaggedMessageSuccess(false);
```

Comment:
Reached the maximum records that can fit inside the publisher buffer, signal that we have not completed retrieving all tagged records.